myproxy/source/myproxy.c

```diff
@@ -554,21 +553,16 @@ myproxy_bootstrap_trust(myproxy_socket_attrs_t *attrs)
     SSL_CTX_set_options(ctx, SSL_OP_NO_SSLv2 | SSL_OP_NO_SSLv3);
     #endif
 
-    if (!(sbio = BIO_new_ssl_connect(ctx))) goto error;
+    if (!(sbio = BIO_new_ssl(ctx, 1))) goto error;
     if ( (sockfd = get_connected_myproxy_host_socket(
                    attrs->pshost, attrs->psport) ) < 0) {
         goto error;
     }
     BIO_get_ssl(sbio, &ssl);
-    char chport[6];
-    snprintf(chport, sizeof(chport), "%d", attrs->psport);
-    BIO_set_conn_port(sbio, chport);
-    BIO_set_conn_hostname(sbio, attrs->pshost);
-    BIO_set_fd(sbio, sockfd, 0);
+    SSL_set_fd(ssl, sockfd);
 
     if (BIO_do_handshake(sbio) <= 0) goto error;
     BIO_write(sbio, "0", 1);    /* GSI deleg flag */
-    if (BIO_flush(sbio) <= 0) goto error;
```

Comment:
As the return value of `BIO_write()` is not checked, maybe line 571 (or 566 if kept) was meant to make sure that the flag reaches the other side of the connection.

Or maybe change 570 (now 565) to:
```
if (BIO_write(sbio, "0", 1) < 1) goto error ;    /* GSI deleg flag */
```
...?

Reply:
I added a check of the return value of BIO_write as suggested.

Reply:
Thanks.